mono/metadata/assembly.c

```diff
@@ -1828,6 +1827,24 @@ mono_assembly_load_friends (MonoAssembly* ass)
 	mono_assemblies_unlock ();
 }
 
+struct HasReferenceAssemblyAttributeIterData {
+	gboolean has_attr;
+};
+
+static gboolean
+has_reference_assembly_attribute_iterator (MonoImage *image, const MonoAssemblyName *aname, const char *nspace, const char *name, gpointer user_data)
```

Comment:
I'll change it to just get the typeref scope token.

I'm still holding out hope of making this into a general-purpose API in which case it seems like the iterator ought to get:
  1. enough info to construct the assembly-qualified type name
  2. the method token
  3. (not sure if this is needed) enough info so that the iterator can decode the arguments blob if it wants to.